crates/aptos/src/account/key_rotation.rs

```diff
@@ -222,17 +222,48 @@ impl CliCommand<RotateSummary> for RotateKey {
             return Err(CliError::AbortedError);
         }
 
-        let mut profile_config = ProfileConfig {
-            private_key: Some(new_private_key.clone()),
-            public_key: Some(new_private_key.public_key()),
-            account: Some(sender_address),
-            ..self.txn_options.profile_options.profile()?
+        let profile_config = match self.txn_options.profile_options.profile() {
+            Ok(profile) => ProfileConfig {
+                private_key: Some(new_private_key.clone()),
+                public_key: Some(new_private_key.public_key()),
+                account: Some(sender_address),
+                ..profile
+            },
+            Err(_) => {
+                let (rest_url, faucet_url) = if let Some(url) = self.txn_options.rest_options.url {
+                    let url_str = url.to_string();
+                    if url_str.contains("testnet") {
+                        (
+                            Some(url_str),
+                            Some("https://faucet.testnet.aptoslabs.com".to_string()),
+                        )
+                    } else if url_str.contains("devnet") {
+                        (
+                            Some(url_str),
+                            Some("https://faucet.devnet.aptoslabs.com".to_string()),
+                        )
+                    } else if url_str.contains("localhost") {
+                        (Some(url_str), Some("http://localhost:8081".to_string()))
+                    } else {
+                        // no faucet for mainnet
+                        (Some(url.to_string()), None)
+                    }
```

Comment:
This is super fragile, and you won't be able to tell if a custom URL is mainnet, local etc.

It should just put nothing in the faucet URL for now, and someone can configure it afterwards.

Reply:
Hmm, are we ok with an profile created with an empty faucet url for devnet and testnet?

Reply:
Yeah, it's a slightly unfavorable situation, but it's preferred over behavior that doesn't work as expected (funding a different network)

Reply:
Make sense.

Another idea, how about if I update the check from `.contains()` to `==`, this way it will only add the [devnet, testnet] faucet url when rest_url is actually matching [devnet, testnet]. Everything else will be set to `None`

Reply:
`devnet` expands to `https://fullnode.devnet.aptoslabs.com` in the config.yaml so maybe just match the fully qualified node url?